common/src/main/java/com/gregtechceu/gtceu/utils/GTUtil.java

```diff
@@ -335,4 +338,24 @@ public static FluidStack getFluidFromContainer(Object ingredient) {
         return null;
     }
 
+    public static boolean canSeeSunClearly(Level world, BlockPos blockPos) {
+        if (!world.canSeeSky(blockPos.above())) {
+            return false;
+        }
+        Biome biome = world.getBiome(blockPos.above()).value();
+        if (world.isRaining()) {
+            if (biome.warmEnoughToRain(blockPos.above()) || biome.coldEnoughToSnow(blockPos.above())) {
+                return false;
+            }
+        }
+
+
+        if (world.getBiome(blockPos.above()).is(Biomes.THE_END.registry())) {
```

Comment:
ehhh
didn't realise last time but this should check for `BiomeTags#IS_END` instead, to match all biomes instead of just the main island.

Reply:
don(?)